scala/private/common_attributes.bzl

```diff
@@ -90,6 +90,30 @@ implicit_deps = {
         default = Label("@io_bazel_rules_scala//src/java/io/bazel/rulesscala/scalac"),
         allow_files = True,
     ),
+    "_scalac_before_2_12_13": attr.label(
```

Comment:
Hi, @mateuszkuta256, thanks for PR. Does that mean than every scala version needs to be setup in workspace for this toolchain to work? ie all flavours of scalac worker needs to be compiled?

Reply:
hey, let me explain this change in detail, as I am searching for an alternative solution anyway...
sources for `scalac` are defined [here](https://github.com/mateuszkuta256/rules_scala/blob/a02f738f84af2cfbf66520ceb437bc2f6f95dd6b/src/java/io/bazel/rulesscala/scalac/BUILD#L104)
these are conditional upon: `if SCALA_MAJOR_VERSION.startswith("2") `
so I simply made a few `scalac_...` targets, separate for 2.11, 2 and 3
would love to keep a single `scalac`, but missing an idea of how to achieve that...
if we don't specify e.g. scala 3.3.1 in a config setting then I believe _scalac3 is skipped (?)
at least the build is successful without complaining that scala3 deps are missing

Reply:
Right, I see. At this point it's just a label and magic happens on action when you select `scalac`.

As for alternatives I was playing with toolchain setup where each scala version is encapsulated/scoped in it's own repository and compiles it's own `scalac` (similar to `remote_java_repository`). https://github.com/simuons/rules_scala/commit/919471ae6d04811b3b07a5ed134278f5681c71e9 Still not sure where this leads.

Reply:
I see you got rid of the circular dependency between toolchain and `scalac` caused by `toolchain_deps:scala_compile_classpath` dep. It is something that I was missing in my first drafts and allows to apply nice solutions 👍 
in the meantime, I tested something like this https://github.com/mateuszkuta256/rules_scala/commit/f247cc8a8d0ad249d78743f2059106a91a10a4c9
`_scalac` is removed from implicit attributes and attached to `scala_library` and other rules. Then we can make a version specific implementations, e.g.
`scala_3_3_binary = make_scala_binary(scala_version = "3.3.1")`
since you found a way to remove circular dependencies, the same thing can be applied directly on the toolchain [here](https://github.com/mateuszkuta256/rules_scala/blob/d8259b1907b0429cf4d4161088d5cbfbc2600251/scala/scala_toolchain.bzl#L177)
from the user perspective, it's even nicer, because can use regular scala_library without making version specific one